compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -387,3 +388,41 @@ ireeGPUGetSingleSubgroupLayout(MlirAttribute attr, uint32_t fragment) {
   result.element = wrap(builder.getI64ArrayAttr(layout.element));
   return result;
 }
+
+ireeGPUTargetInfo
+ireeHALExecutableTargetAttrGetGPUTargetInfo(MlirAttribute attr) {
+  assert(!mlirAttributeIsNull(attr) && "attr cannot be null");
+  auto executableTargetAttr =
+      llvm::cast<mlir::iree_compiler::IREE::HAL::ExecutableTargetAttr>(
+          unwrap(attr));
+
+  assert(executableTargetAttr && "attr is not a HAL::ExecutableTargetAttr");
+
+  ireeGPUTargetInfo targetInfo = {};
+  auto context = executableTargetAttr.getContext();
+  auto gpuTargetAttr =
```

Comment:
Please spell out the types here and below, where the type is not obvious based on the RHS only. See https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable